docs/integrations/azure/discovery.md

```diff
@@ -80,9 +82,10 @@ The parameters available are:
   - **`scope`** _(optional)_:
     `'global'` or `'local'`. Sets the scope of concurrency control.
 
-_Note:_ the path parameter follows the same rules as the search on Azure DevOps
-web interface. For more details visit the
-[official search documentation](https://docs.microsoft.com/en-us/azure/devops/project/search/get-started-search?view=azure-devops).
+_Note:_
+
+- The path parameter follows the same rules as the search on Azure DevOps web interface. For more details visit the [official search documentation](https://docs.microsoft.com/en-us/azure/devops/project/search/get-started-search?view=azure-devops).
+- The branch parameters need that the branch desired it`s added on 'Searchable branches' on Azure DevOps Repositories
```

Comment:
Please help me understand how this works in Azure.
Is it correct that configuring the `branch` will make sure that the discovery provider only look in the `development` branch for example?
If that's the behaviour I don't think we need to have an additional line here as a note.

Reply:
The note about "Searchable branches" is because Azure CodeSearch only searches in the branch added on "Searchable Branches". The default is to only have the default branch (main/master)

So, if the branch provided by the backstage.io config isn`t added in the "Searchable branches", the Code Search will not find the catalog files.

Currently, even if I add the desired branch on "Searchable branches", the Code Search discovery will find the file, but cannot retrieve it because it`s pointed to the default branch. My PR is to enable to retrieve of the file in the provided branch.

![image](https://user-images.githubusercontent.com/5731429/223708046-d0e3ad2b-3caa-4453-8b4b-5ae1e12f8c62.png)

Reply:
Given that these extra steps need to be taken to get this to work I think they should be included in the documentation. The link - https://docs.microsoft.com/en-us/azure/devops/project/search/get-started-search?view=azure-devops - doesn't seem to show how to do this

Reply:
@awanlin somenthin like that:

```markdown
- To use branch parameters, it is necessary that the desired branch be added to the "Searchable branches" list within Azure DevOps Repositories. To do this, follow the instructions below:
1. Access your Azure DevOps and open the repository in which you want to add the branch.
2. Click on "Settings" in the lower left corner of the screen.
3. Select the "Options" option in the left navigation bar.
4. In the "Searchable branches" section, click on the "Add" button to add a new branch.
5. In the window that appears, enter the name of the branch you want to add and click "Add".
6. The added branch will now appear in the "Searchable branches" list.
```

Reply:
Yes, this is perfect, thanks for taking the time to write this up. Will save many people in the long run!